src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,960 @@
+        self.__assert_status_file_data_multi_exception_errors(complete_substatus_file_data, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_ERROR, patch_count_installation,
+            count_errors_detail=5, substatus_index=1)
+
+        # Assert assessment truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            truncated_substatus_file_data = json.load(file_handle)
+
+        # Assert length of truncated patches < length of completed patches
+        assessment_truncated_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data)
+        installation_truncated_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data, substatus_index=1)
+        self.assertTrue(len(assessment_truncated_msg["patches"]) + len(installation_truncated_msg["patches"]) < patch_count_assessment + patch_count_installation)
+
+        # Assert assessment truncated
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_ASSESSMENT_SUMMARY,
+            Constants.STATUS_WARNING, patch_count_assessment, assessment_truncated_msg)
+
+        # Assert all assessment fields in the message json are equal in both status files
+        self.__assert_assessment_truncated_msg_fields(assessment_msg_with_errors, assessment_truncated_msg)
+
+        # Assert installation truncated status file with multi exceptions
+        self.__assert_truncated_status_multi_exceptions(complete_substatus_file_data, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_ERROR, patch_count_installation, errors_count=5,
+            substatus_index=1)
+
+        # Assert all installation fields in the message json are equal in both status files
+        self.__assert_installation_truncated_msg_fields(installation_msg_with_errors, installation_truncated_msg)
+
+    def test_log_truncated_patches_assert_not_truncated(self):
+        """ Assert no truncation is performed on assessment/installation patches.
+        Output:
+        'Count of patches removed from: [Assessment=0] [Installation=0] log message is called.' """
+
+        # Set up temp file stor all log information and set sys.stdout point to it
+        self.__create_tmp_file_for_log_and_set_stdout()
+
+        # Assert no truncation log output
+        patch_count = 500
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        self.runtime.status_handler.log_truncated_patches()
+        self.__read_tmp_log_and_assert("Count of patches removed from: [Assessment=0] [Installation=0]")
+
+        # Reset sys.stdout, close and delete tmp
+        self.__remove_tmp_file_reset_stdout()
```

Comment:
Wasn't necessary but a very good test case implementation. Think of combining this with the other tests. Wherein setting and asserting truncation happens as it is now, at the end of the test case, set stdout, call __read_tmp_log_and_assert().

That would remove all of these tests(this one and below) written just to validate the log

Reply:
@feng-j678 what is the update on this?